src/IMangrove.sol

```diff
@@ -62,118 +101,334 @@ interface IMangrove is HasMgvEvents {
     uint maxGasreqForFailingOffers
   ) external returns (uint takerGot, uint takerGave, uint bounty, uint fee);
 
+  ///@notice Performs a market order on a specified offer list taking offers up to a limit price.
+  ///@param olKey The offer list key given by (maker) `outbound`, (maker) `inbound`, and `tickSpacing`.
+  ///@param takerWants The amount the taker wants. This is used along with `takerGives` to derive a max price (`maxTick`) note that some precision is lost to rounding.
```

Comment:
Are there rounding errors happening here? I was under the impression that we would always derive a `maxTick` such that `log_1.0001(takerGives/takerWants) <= maxTick`.

```suggestion
  ///@param takerWants The amount the taker wants. This is used along with `takerGives` to derive a max price (`maxTick`) which is the lowest allowed tick in the offer list such that `log_1.0001(takerGives/takerWants) <= maxTick`.
```